### ISSUE TITLE
[use-lifecycle-interface] False positive when class extends class which implements lifecycle interface

### ISSUE THREAD

**Issue by amitbeck** [angular-eslint/angular-eslint#866]
**Description and reproduction of the issue**

```JSON
{
  "extends": [
    "plugin:@angular-eslint/recommended"
  ],
  "plugins": ["@typescript-eslint", "@angular-eslint"],
}
```

```TS
// your repro code case
@Directive()
export class FoobarBase implements OnDestroy {
  // ...

  ngOnDestroy(): void {
    /* some base logic here */
  }
}

@Component({...})
export class FoobarComponent extends FoobarBase {
  // ...

  ngOnDestroy(): void { // doesn't work with `override` neither
    super.ngOnDestroy();

    /* some concrete logic here */
  }
}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/eslint-plugin`    | `12.7.0` |
| `@typescript-eslint/parser`        | `4.28.5` |
| `ESLint`                           | `7.26.0` |
| `node`                             | `16.13.1` |

```sh
# Please run `npx ng version` in your project and paste the full output here:

Angular CLI: 12.2.13
Node: 16.13.1 (Unsupported)
Package Manager: npm 8.1.2
OS: darwin x64

Angular: 12.2.13
... animations, cdk, cdk-experimental, cli, common, compiler
... compiler-cli, core, forms, language-service, localize
... material, platform-browser, platform-browser-dynamic, router
... service-worker, upgrade

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1202.13
@angular-devkit/build-angular   12.2.13
@angular-devkit/core            12.2.13
@angular-devkit/schematics      12.2.13
@schematics/angular             12.2.13
rxjs                            6.5.5
typescript                      4.3.5
    
Warning: The current version of Node (16.13.1) is not supported by Angular.
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by balintsteinbach** [comment]
Same issue here.

Angular CLI: 14.2.10
Node: 14.17.3
Package Manager: npm 9.2.0 
OS: linux x64

├── @typescript-eslint/eslint-plugin@5.43.0
├── @typescript-eslint/parser@5.43.0
├── eslint-plugin-import@2.26.0
├── eslint-plugin-jsdoc@39.6.4
├── eslint-plugin-prefer-arrow@1.2.3
├── eslint@8.30.0

**Comment by reduckted** [comment]
🤔 I had a look at whether the type checker could be used to get the interfaces implemented by base types. I may be looking in the wrong place, but it doesn't look like it's possible. Using the example above, the base types of `FoobarComponent` are `[ FoobarBase ]` and the base types of `FoobarBase` are `[ ]`.

Maybe an easy solution to this problem is to assume that if the method is marked as `override`, then the base class will implement the required interface, so the rule could just skip any methods marked as `override`?
